### PR TITLE
options/posix: Stub get and setdomainname

### PR DESCRIPTION
**Pull request by Dennisbonke**
Split off from #544

Part of the mlibc LFS project.

### CODE REVIEW

**Review by avdgrinten — APPROVED** [review]

**Comment by 64** [comment]
These should probably live in `options/glibc` instead, right? But that'd probably require some other changes anyway, so maybe we should make an issue for that and leave it to a future PR.

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
Correct, but it's silly to stub this.

```
       On most Linux architectures (including x86), there is no getdomainname() system call; instead, glibc implements getdomainname() as a  library  func‐
       tion that returns a copy of the domainname field returned from a call to uname(2).
```
... and `setdomainname` is a syscall.

File: options/posix/generic/unistd-stubs.cpp [hunk 1]
```diff
@@ -1122,6 +1122,7 @@ char *ctermid(char *s) {
 	return s ? strcpy(s, "/dev/tty") : const_cast<char *>("/dev/tty");
 }
 
+<<<<<<< HEAD
```
Comment (ArsenArsen) [inline options/posix/generic/unistd-stubs.cpp:1125]: Merge markers

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
The combined implementation for these two functions and their Linux sysdeps is under 50LoC, boilerplate included

**Comment by Dennisbonke** [comment]
> 

Unfortunately we don't have domainname in the mlibc abi, so that is blocked on abi-break. I move to implement this at a later point, and merge the stubs instead for now/

**Comment by Dennisbonke** [comment]
> Didn't we just merge an abi-break because a struct was incorrect on Linux?

Yes, so the Linux abi is fine, but the mlibc abi is not fine until abi-break is merged into master. I don't want to wait on that.

**Comment by ArsenArsen** [comment]
The struct access would be inside sysdeps to accomodate systems that have a getdomainname syscall anyway, and hence does not rely on the mlibc ABI on Linux

**Comment by Geertiebear** [comment]
I don't think an implementation of these is required to merge this PR, that can always be done in a future PR. But the merge marker needs to be removed for it to build of course.

**Review by 64 — APPROVED** [review]

File: options/posix/include/unistd.h [hunk 2]
```diff
@@ -252,6 +252,10 @@ int pipe2(int *pipefd, int flags);
 int setresuid(uid_t ruid, uid_t euid, uid_t suid);
 int setresgid(gid_t rgid, gid_t egid, gid_t sgid);
 
+// Glibc extensions
```
Comment (ArsenArsen) [inline options/posix/include/unistd.h:255]: line comment

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Pull request merged**